# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/nsCoreUtils.cpp
# Commit: 09de143215a1
# Full Hash: 09de143215a13a22dcb2cd53d4ed1dfb60480c82
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-09-30 02:48 +0300
# Regressor Bug: 1987945
# File Overlap Count: 2
# Overlapping Files: accessible/base/nsCoreUtils.cpp, accessible/tests/browser/relations/browser_anchor_positioning.js
# Description:
#   Revert "Bug 1987945: apply code formatting via Lando" for causing bustages at DocAccessible.cpp.
# ==============================================================================

diff -r 826eb4ac00dd -r 09de143215a1 accessible/base/nsCoreUtils.cpp
--- a/accessible/base/nsCoreUtils.cpp	Tue Sep 30 02:35:22 2025 +0300
+++ b/accessible/base/nsCoreUtils.cpp	Tue Sep 30 02:48:50 2025 +0300
@@ -674,70 +674,3 @@
       nsIFrame::TrailingWhitespace::Trim);
   return text.mString.IsEmpty();
 }
-
-nsIFrame* nsCoreUtils::GetAnchorForPositionedFrame(
-    const PresShell* aPresShell, const nsIFrame* aPositionedFrame) {
-  if (!aPositionedFrame ||
-      !aPositionedFrame->Style()->HasAnchorPosReference()) {
-    return nullptr;
-  }
-
-  const nsAtom* anchorName = nullptr;
-  AnchorPosReferenceData* referencedAnchors =
-      aPositionedFrame->GetProperty(nsIFrame::AnchorPosReferences());
-
-  if (!referencedAnchors) {
-    return nullptr;
-  }
-
-  for (auto& entry : *referencedAnchors) {
-    if (entry.GetData().isNothing()) {
-      continue;
-    }
-
-    if (anchorName && entry.GetKey() != anchorName) {
-      // Multiple anchors referenced.
-      return nullptr;
-    }
-
-    anchorName = entry.GetKey();
-  }
-
-  return anchorName
-             ? aPresShell->GetAnchorPosAnchor(anchorName, aPositionedFrame)
-             : nullptr;
-}
-
-nsIFrame* nsCoreUtils::GetPositionedFrameForAnchor(
-    const PresShell* aPresShell, const nsIFrame* aAnchorFrame) {
-  if (!aAnchorFrame) {
-    return nullptr;
-  }
-
-  nsIFrame* positionedFrame = nullptr;
-  const auto* styleDisp = aAnchorFrame->StyleDisplay();
-  if (styleDisp->HasAnchorName()) {
-    for (auto& name : styleDisp->mAnchorName.AsSpan()) {
-      for (nsIFrame* frame : aPresShell->GetAnchorPosPositioned()) {
-        // Bug 1990069: We need to iterate over all positioned frames in doc and
-        // check their referenced anchors because we don't store reverse mapping
-        // from anchor to positioned frame.
-        const auto* referencedAnchors =
-            frame->GetProperty(nsIFrame::AnchorPosReferences());
-        const auto* data = referencedAnchors->Lookup(name.AsAtom());
-        if (data && *data && data->ref().mOrigin) {
-          if (aAnchorFrame ==
-              aPresShell->GetAnchorPosAnchor(name.AsAtom(), frame)) {
-            if (positionedFrame) {
-              // Multiple positioned frames reference this anchor.
-              return nullptr;
-            }
-            positionedFrame = frame;
-          }
-        }
-      }
-    }
-  }
-
-  return positionedFrame;
-}